interp/interp_test.go

```diff
@@ -1038,3 +1041,34 @@ func TestRunnerOpts(t *testing.T) {
 		})
 	}
 }
+
+func TestContext(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+	defer cancel()
+
+	p, err := syntax.Parse(strings.NewReader("while true; do true; done;"), "", 0)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	buff := bytes.NewBuffer(nil)
```

Comment:
In fact, I think you can just ignore Stdout (or if it crashes, `ioutil.Discard` and I'll fix that later)